### PR TITLE
Plays food processing sound functionality

### LINKED ISSUE #5906: Add sound to food processors.
Right now food processors are fully functional gameplay wise, but are lacking a sound and the shaking animation that occurs while it's actively processing food.

The basic functions for the animation are already in FoodProcessor.cs, but due to a LeanTween build error it was commented out. 

Right now there is a syncvar that changes to true while the processor is on and false while the processor is off, and is used to play a looping sound, which was taken from Microwave.cs. However in light of the new sound addressable system both food processors and microwaves do not play sounds while active, probably meaning the function needs to be adjusted to fit the new system.

- [x] add shaking animation
- [x] add sound effect

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #5906

### CODE REVIEW

**Comment by corp-0** [comment]
amazing

**Comment by corp-0** [comment]
shouldn't the sound loop for as long as the animation plays?

**Comment by MaxIsJoe** [comment]
> shouldn't the sound loop for as long as the animation plays?

the sound isn't designed to do that at the moment, I'll probably update it sometime later to allow that.

**Comment by ThatDan123** [comment]
Line 261 already does the sound triggering.

> the sound isn't designed to do that at the moment, I'll probably update it sometime later to allow that.

The sound is also set to loop

**Comment by ThatDan123** [comment]
It might just be that #5906 just wasnt closed when the sound was added or its bugged now?

**Comment by MaxIsJoe** [comment]
> It might just be that #5906 just wasnt closed when the sound was added or its bugged now?

it's bugged now

**Comment by corp-0** [comment]
welp if it is bugged then we need to fix that, not add another sound trigger.

**Comment by MaxIsJoe** [comment]
> welp if it is bugged then we need to fix that, not add another sound trigger.

did you just block out what i said? you can't loop the sound because it's not designed to be looped so it will sound ugly.

**Comment by MaxIsJoe** [comment]
you'll need to separate the juicer sound into three different audio files and make sure that all three seamlessly blend together when played.
I don't want to do that in this PR, I just want to get the sound working until i feel like working on editing the audio to allow that.

**Comment by ThatDan123** [comment]
> did you just block out what i said? you can't loop the sound because it's not designed to be looped so it will sound ugly.

The existing method to play the sound doesnt auto loop it, thats done in the audio source sound file. 

So you should only need to fix why the current play method isnt working if you dont want a looped sound

**Comment by MaxIsJoe** [comment]
I edited the files and made them loopable, problem is now is trying to find a way for the audio to stay in sync with the animation duration because we're using `Time.delta` to handle it.

File: UnityProject/Assets/Scripts/Objects/Machines/FoodProcessor.cs [hunk 1]
```diff
@@ -253,17 +262,22 @@ private void RpcShake(float duration, float distance, float delayBetweenShakes)
...
+		private async void HandleAudio(float duration)
 		{
-			if (newState)
-			{
-				runLoopGUID = Guid.NewGuid().ToString();
-				SoundManager.PlayAtPositionAttached(RunningAudio, registerTile.WorldPosition, gameObject, runLoopGUID);
-			}
-			else
+			if(CustomNetworkManager.IsServer == false) return;
+			Debug.Log($"{duration} - {(int)fullSource.Result.AudioSource.clip.length}");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Machines/FoodProcessor.cs]:
```suggestion
```

**Comment by MaxIsJoe** [comment]
Alright so i have no idea how to fix that minor issue so I'll just leave a todo comment

also im going to solve the merge conflicts tomorrow because that requires me rebuilding addressables

**Comment by MaxIsJoe** [comment]
closing this because i can't properly solve merge conflicts.
opening a new PR

**Pull request closed without merging**